dev-utilities/hoistUseStatements.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Moves `@use` statements before data injected via vite’s `css.preprocessorOptions.scss.additionalData` option which avoids the following Sass error:
+ *
+ * ```
+ * @use rules must be written before any other rules.
+ * ```
+ *
+ * Source: https://github.com/vitejs/vite/discussions/6361#discussioncomment-6577205
```

Comment:
do we have an issue to track this? so we can remove this hack when it's fixed upstream?

Reply:
This isn’t really a bug though.

From the perspective of Sass, it’s probably a reasonable design decision to force `@use` statements to be the first in a file.

From the perspective of a build tool, it’s reasonable to not mess with preprocessing other than giving users the option to alter files on their own responsibility (i.e. what we’re making use of here).

So, this is an unfortunate compromise. The alternative would be to not prefix all Sass files with the variables import but that would mean that we would need to manually import the variables in all our `<style lang="scss">` blocks. A hassle I’d like to avoid.